<a href="https://colab.research.google.com/github/jadercaro/Mercado-inmobiliario-ruso-de-Sberbank/blob/main/02_Preprocesamiento.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Librerías principales

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import os
import glob
from scipy.stats import norm
from sklearn.preprocessing import StandardScaler
from scipy import stats
import warnings
warnings.filterwarnings('ignore') # Para evitar los molestos avisos.
%matplotlib inline
import missingno as msno

#Cargamos la base de datos construida
La base de datos original esta alojada en la carpeta **/Dataset** en el github, lo que hacemos es clonar el repositorio directamente desde el github y luego se tendrá que descomprimir

In [2]:
!git clone https://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@github.com/jadercaro/Mercado-inmobiliario-ruso-de-Sberbank/
os.chdir('/content/Mercado-inmobiliario-ruso-de-Sberbank/Dataset/')
!unzip dataset_train.zip

Cloning into 'Mercado-inmobiliario-ruso-de-Sberbank'...
remote: Enumerating objects: 91, done.
remote: Counting objects: 100% (91/91), done.
remote: Compressing objects: 100% (86/86), done.
remote: Total 91 (delta 29), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (91/91), 66.88 MiB | 14.94 MiB/s, done.
Resolving deltas: 100% (29/29), done.
Archive:  dataset_train.zip
  inflating: dataset_train.csv       


In [4]:
dataset = pd.read_csv('dataset_train.csv')

#1.Cantidad de datos faltantes
Veremos la cantidad de datos faltantes que tenemos y el restante lo simularemos eliminando de manera aleatoria estos datos, con el fin de poder lograr el objetivo de tener el 5% de los datos nulos como parte del reto competitivo de este proyecto.

In [13]:
nulos_final = dataset.isnull().sum().sum()
print("{:.2}%".format(nulos_final*100/(len(dataset)*len(dataset.columns))))
datos_nulos_faltantes=int((len(dataset)*len(dataset.columns)*0.0495)-nulos_final)
print(datos_nulos_faltantes +1)

4.2%
85214


La Cantidad de datos nulos hasta el momento es de 4.2% aproximadamente, para lograr obtener el 5% de datos nulos es necesario eliminar aleatoriamente **85214 registros**